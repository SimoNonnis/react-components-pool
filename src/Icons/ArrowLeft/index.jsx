import React from 'react';
import PropTypes from 'prop-types';

/**
 * ArrowLeft Icon.
 * @param {number} width
 * @param {number} height
 * @param {string} fill
 * @param {string} className
 * @param {string} classNamePath
 * @returns {React.Component}
 */
function ArrowLeft({ width, height, fill, className, classNamePath }) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 20 20"
      width={width}
      height={height}
      className={className}
    >
      <title>ArrowLeft Icon</title>

      <path
        className={classNamePath}
        fill={fill}
        d="M14.52,18.48a.5.5,0,0,0,.33-.88L6.23,10,14.85,2.4a.5.5,0,1,0-.66-.75l-9,8a.5.5,0,0,0,0,.75l9,8A.5.5,0,0,0,14.52,18.48Z"
      />
    </svg>
  );
}

ArrowLeft.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  fill: PropTypes.string,
  className: PropTypes.string,
  classNamePath: PropTypes.string,
};

ArrowLeft.defaultProps = {
  width: 40,
  height: 40,
  fill: 'currentColor',
  className: 'arrowLeftIcon',
  classNamePath: 'arrowLeftIconPath',
};

export default ArrowLeft;
