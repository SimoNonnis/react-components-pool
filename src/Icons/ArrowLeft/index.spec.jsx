/* eslint-env jest */
import React from 'react';

import ArrowLeft from './';

describe('Test <ArrowLeft /> icon', () => {
  it('should render ArrowLeft icon correctly with default props', () => {
    const wrapper = shallow(<ArrowLeft />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render ArrowLeft icon correctly with props', () => {
    const wrapper = shallow(
      <ArrowLeft fill="#a4a1a1" className="testClassname" height={30} width={30} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
