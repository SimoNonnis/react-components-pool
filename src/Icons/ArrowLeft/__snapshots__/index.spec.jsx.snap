// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ArrowLeft /> icon should render ArrowLeft icon correctly with default props 1`] = `
<svg
  className="arrowLeftIcon"
  height={40}
  viewBox="0 0 20 20"
  width={40}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    ArrowLeft Icon
  </title>
  <path
    className="arrowLeftIconPath"
    d="M14.52,18.48a.5.5,0,0,0,.33-.88L6.23,10,14.85,2.4a.5.5,0,1,0-.66-.75l-9,8a.5.5,0,0,0,0,.75l9,8A.5.5,0,0,0,14.52,18.48Z"
    fill="currentColor"
  />
</svg>
`;

exports[`Test <ArrowLeft /> icon should render ArrowLeft icon correctly with props 1`] = `
<svg
  className="testClassname"
  height={30}
  viewBox="0 0 20 20"
  width={30}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    ArrowLeft Icon
  </title>
  <path
    className="arrowLeftIconPath"
    d="M14.52,18.48a.5.5,0,0,0,.33-.88L6.23,10,14.85,2.4a.5.5,0,1,0-.66-.75l-9,8a.5.5,0,0,0,0,.75l9,8A.5.5,0,0,0,14.52,18.48Z"
    fill="#a4a1a1"
  />
</svg>
`;
